designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -102,23 +106,44 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
     (for {
       validation <- EitherT(validateProcess(localProcess))
       _ <- EitherT.fromEither[Future](if (validation.errors != ValidationErrors.success) Left[EspError, Unit](MigrationValidationError(validation.errors)) else Right(()))
-      _ <- createRemoteProcessIfNotExist(localProcess, category)
+      processExistsOnRemote <- checkIfProcessExistsOnRemote(localProcess)
+      _ <- if (processExistsOnRemote) {
+        isArchivedOnRemote(localProcess)
+      } else {
+        createRemoteProcess(localProcess, category)
+      }
       _ <- EitherT.right[EspError](saveProcess(localProcess, UpdateProcessComment(s"Scenario migrated from $environmentId by ${loggedUser.username}")))
     } yield ()).value
   }
 
-  private def createRemoteProcessIfNotExist(localProcess: DisplayableProcess, category: String)
-                                           (implicit ec: ExecutionContext): EitherT[Future, EspError, Unit] = {
+  private def checkIfProcessExistsOnRemote(localProcess: DisplayableProcess)
```

Comment:
changed it, also instead of NotFoundError chose RemoteEnvironmentCommunicationError in pattern matching because it did not work otherwise as methods in this class throw specific error which can be 404 but not always so could not make it extend NotFoundError